app/controllers/patients_controller.rb

```diff
@@ -249,53 +249,170 @@ def update
+      render(json: { error: error_message }, status: :forbidden) && return
+    end
 
-    # Make sure all household ids are within jurisdiction
-    redirect_to(root_url) && return if patients_to_update.any? do |patient_id|
-      !current_user_patients.exists?(patient_id)
+    # If the current patients is a HoH, they can't be removed.
+    if current_patient.head_of_household
+      error_message = 'Remove from household  action failed: Monitoree is a head of household. Please refresh.'
+      render(json: { error: error_message }, status: :bad_request) && return
     end
 
-    # Change all of the patients in the household, including the current patient to have new_hoh_id as the responder
-    current_user_patients.where(id: patients_to_update).each do |patient|
-      patient.update!(responder_id: new_hoh_id)
+    # ----- Record Updates -----
+    old_hoh = current_user.get_patient(current_patient.responder_id)
+
+    # Update the record
+    updated = current_patient.update(responder_id: current_patient.id)
+
+    if !updated
+      error_message = 'Remove from household action failed: Monitoree was unable to be be updated.'
+      render(json: { error: error_message, status: :forbidden }) && return
+    else
+      # Create history item for old HoH
+      comment = "User removed dependent monitoree with ID #{current_patient.id} from the household. #{old_hoh.first_name} #{old_hoh.last_name}"\
+                ' will no longer be responsible for handling their reporting.'
+      History.monitoring_change(patient: old_hoh, created_by: current_user.email, comment: comment)
+
+      # Create history item on current patient
+      comment = "User removed monitoree from a household. Monitoree with ID #{old_hoh.id} will"\
+                ' no longer be responsible for handling their reporting.'
+      History.monitoring_change(patient: current_patient, created_by: current_user.email, comment: comment)
+    end
+  end
+
+  # Changes the HoH of a household
+  def update_hoh
+    new_hoh_id = params.permit(:new_hoh_id)[:new_hoh_id]&.to_i
+    current_patient_id = params.permit(:id)[:id]
+    household_ids = params[:household_ids] || []
+
+    current_patient = current_user.get_patient(current_patient_id)
+    new_hoh = current_user.get_patient(new_hoh_id)
+    current_user_patients = current_user.patients
+
+    # NOTE: This includes the new HoH
+    patient_ids_to_update = household_ids + [current_patient_id]
+
+    # ----- Error Checking -----
+
+    # Check to make sure selected HoH record exists.
+    unless Patient.where(purged: false).exists?(new_hoh_id)
+      error_message = "Change head of household action failed: Selected Head of Household with ID #{new_hoh_id} does not exist."
+      render(json: { error: error_message }, status: :forbidden) && return
     end
+
+    # Check to make sure user has access to update all of these records.
+    unless current_user_patients.where(id: patient_ids_to_update).count == patient_ids_to_update.length
```

Comment:
not completely sure if this will make a difference in performance because i'm not sure if these records will be loaded by the time this is executed, but I think it wouldn't hurt to use `size` instead of `count` here: https://stackoverflow.com/questions/6083219/activerecord-size-vs-count

Reply:
Agreed, updated here: a939bdc